oauthlib/openid/connect/core/request_validator.py

```diff
@@ -38,6 +38,31 @@ def get_authorization_code_scopes(self, client_id, code, redirect_uri, request):
         """
         raise NotImplementedError('Subclasses must implement this method.')
 
+    def get_authorization_code_nonce(self, client_id, code, redirect_uri, request):
+        """ Extracts nonce from saved authorization code.
+
+        If present in the Authentication Request, Authorization
+        Servers MUST include a nonce Claim in the ID Token with the
+        Claim Value being the nonce value sent in the Authentication
+        Request. Authorization Servers SHOULD perform no other
+        processing on nonce values used. The nonce value is a
+        case-sensitive string.
+
+        Only code param should be sufficient to retrieve grant code from
+        any storage you are using, `client_id` and `redirect_uri` can gave a
```

Comment:
Typo "gave/have".

Was wondering why `client_id` can be unset, and also why it is then "" instead `None`.

Reply:
Thx, it was a wrong assumption from the copy/paste based on the other method. I have corrected that.